pkg/controller/service_control.go

```diff
@@ -130,6 +133,53 @@ func (c *realServiceControl) recordServiceEvent(verb, name, kind string, object
 	}
 }
 
+func (c *realServiceControl) SyncComponentService(tc runtime.Object, newSvc, oldSvc *corev1.Service, isHeadlessSvc bool) (*corev1.Service, error) {
```

Comment:
The name of `isHeadlessSvc` is not accurate. e.g both PD services are passing `false`, and not accurate for DM-master & DM-worker services too.

Reply:
@csuzhangxc good catch! I changed the param name to `setClusterIP` and fix it in dm_master